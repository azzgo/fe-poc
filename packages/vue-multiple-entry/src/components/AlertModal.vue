<template>
  <overlay :visible="visible" @maskClick="maskClosable && $emit('close')">
    <div class="modal">
      <div class="modal-header">{{title}}</div>
      <div class="modal-body"><slot></slot></div>
      <div class="modal-footer">
        <button class="alert-button" @click="$emit('close')">{{alertButtonText}}</button>
      </div>
    </div>
  </overlay>
</template>

<script>
import Overlay from './Overlay.vue'

export default {
  components: {
    Overlay,
  },
  props: {
    visible: {
      type: Boolean,
      default: false,
    },
    maskClosable: {
      type: Boolean,
      default: true,
    },
    title: {
      type: String,
      default: '提示框',
    },
    alertButtonText: {
      type: String,
      default: '确定',
    },
  },
}
</script>

<style lang="scss" scoped>
.modal {
  box-sizing: border-box;
  top: 10%;
  margin: auto;
  background-color: #fff;
  width: 600px;
  border-radius: 5px;
  position: relative;
  z-index: 1000;
}

.modal-header {
  box-sizing: border-box;
  height: 55px;
  padding: 15px;
  font-size: 18px;
  line-height: 1.42857143;
  margin: 0;
}

.modal-body {
  padding: 15px;
}

.modal-footer {
  box-sizing: border-box;
  height: 65px;
  padding: 15px;
  display: flex;
  justify-content: flex-end;
  .alert-button {
    color: #fff;
    background-color: #337ab7;
    border-color: #2e6da4;
    padding: 6px 12px;
    font-size: 14px;
    font-weight: 400;
    line-height: 1.42857143;
  }
}
</style>
