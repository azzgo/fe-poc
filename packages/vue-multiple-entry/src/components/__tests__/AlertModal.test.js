import { shallowMount } from '@vue/test-utils'
import AlertModal from '../AlertModal.vue'

test('AlertModal should work as expect', () => {
  const comp = shallowMount(AlertModal, {
    stubs: {
      overlay: {
        render(h) {
          return h('div', this.$slots.default)
        },
      },
    },
  })

  comp.setProps({
    visible: true,
  })

  expect(comp.html()).toMatchSnapshot()
})
