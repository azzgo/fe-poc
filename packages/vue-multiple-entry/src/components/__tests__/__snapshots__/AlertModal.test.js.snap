// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertModal should work as expect 1`] = `
<div visible="visible">
  <div class="modal">
    <div class="modal-header">提示框</div>
    <div class="modal-body"></div>
    <div class="modal-footer">
      <button class="alert-button">确定</button>
    </div>
  </div>
</div>
`;
